Doxyfile.in

```diff
@@ -0,0 +1,1552 @@
+# Doxyfile 1.6.2
```

Comment:
<Charlton Heston cold-dead-hands meme.>

I find the Makefile many times easier to understand and maintain. CMake is
tolerated because it produces visual studio solutions, but we've lost many
man hours just trying to get it to do the same thing as something that was
easy in the Makefile.

From a Makefile's point of view your project is a DAG with some shell
scripts it needs to run to get from one node to another. If you don't rely
on default rules, it doesn't even know you're compiling something. Build
systems that attempt to understand your source code in some deeper way
usually shit all over themselves when you have generated source,
metaprogramming with multi-stage compilation, weird linkage options, etc.


On Thu, Jun 18, 2015 at 11:25 AM, Dan Liew <notifications@github.com> wrote:

> In Doxyfile.in
> <https://github.com/halide/Halide/pull/825#discussion_r32761840>:
>
> > @@ -0,0 +1,1552 @@
> > +# Doxyfile 1.6.2
>
> It is a new copy :(
>
> I really didn't want to do that but it is essential for out of source
> builds to work (if you take a look in Doxyfile.in you'll see there are a
> bunch of @ substitutions in there).
>
> I'd rather there wasn't two. I'd argue that eventually we'd want to get
> rid of the Makefile build system all together, when we can drop having
> two Doxyfiles.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/halide/Halide/pull/825/files#r32761840>.
>